### PR TITLE
[BUGFIX][COMMUN-435] Add exception handling

### PR DESCRIPTION
**Pull request by plovinda**
Add subscriber for exception handling of suspicious operation exception, add test.

### CODE REVIEW

**Review by andreasfernandez — COMMENTED** [review]
I don't understand how this fixes the issue reported in the ticket.

`Invalid method override "__CONSTRUCT"` looks to me like some internal code in intercept is broken.

**Comment by andreasfernandez** [comment]
> I don't understand how this fixes the issue reported in the ticket.
> 
> `Invalid method override "__CONSTRUCT"` looks to me like some internal code in intercept is broken.

Alright, this happens if a request sent to intercept contains an invalid HTTP method (GET, POST, PATCH, etc).

File: tests/Functional/EventSubscriber/ExceptionSubscriberTest.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    public function testExceptionSubscriberIsCalledOnSuspiciousOperationException(): void
+    {
+        $exceptionSubscriber = new ExceptionSubscriber();
+        $this->dispatcher->addSubscriber($exceptionSubscriber);
+        $kernel = new Kernel('test', true);
+        $kernel->boot();
+        $event = new ExceptionEvent($kernel, new Request(), 2, new SuspiciousOperationException());
+        $this->dispatcher->dispatch($event, 'kernel.exception');
+        $this->assertInstanceOf(SuspiciousOperationException::class, $event->getThrowable());
```
Comment (andreasfernandez) [inline tests/Functional/EventSubscriber/ExceptionSubscriberTest.php:41]: You're testing something you pass into the method two lines above. You should rather test whether `$event->getResponse()` contains the expected data.

**Review by andreasfernandez — CHANGES_REQUESTED** [review]

File: tests/Functional/EventSubscriber/ExceptionSubscriberTest.php (hunk 1 above)
Comment (plovinda) [reply tests/Functional/EventSubscriber/ExceptionSubscriberTest.php]: $event->getResponse returns null. I tried to mock kernal exception to see if ExceptionSubscriber is called. It did not work, so removed the test.

**Comment by andreasfernandez** [comment]
We won't fix this issue. Symfony reacts correctly by sending a "Bad Request".

**Pull request closed without merging**